### PR TITLE
updates the IC5 level description in the handbook

### PR DESCRIPTION
**Pull request by chrispine**
This updates the IC5 level description in our handbook. Previous work on the draft of this was done [in this doc](https://docs.google.com/document/d/16fL4KETgtPPbhux9DkZC6hLbJg6qQO8bw2REuqA8qXs/edit#).

### CODE REVIEW

File: content/departments/engineering/dev/career-development/framework.md [hunk 1]
```diff
@@ -401,23 +400,60 @@ To learn more, see ["Considerations for promotion" in our talent review process]
...
       </td>
     </tr>
     <!-- IC6 -->
     <tr>
       <th id="ic6" scope="row" class="level"><a class="anchor" href="#ic6"></a><abbr title="Individual Contributor">IC</abbr>6</th>
       <td colspan="3" class="tbd">
         <p>
-          We haven't yet finalized the description of this level at Sourcegraph.
-          Like <a href="#ic5">IC5</a>, this is a different <strong>role</strong> than the levels preceding it based not only on performance, but also business need, and (like <a href="#ic5">IC5</a>) what impact at this level looks like may vary more from person to person than at preceding levels.
+          Senior Staff Engineer. We haven't yet finalized the description of this level at Sourcegraph.
```
Comment (felixfbecker) [inline content/departments/engineering/dev/career-development/framework.md:456]: Is it Senior Staff Engineer? Or is it a Principle Engineer? (Or, is the answer we don't know that yet either, in which case I think it shouldn't be stated yet?)

Comment (vrto) [reply content/departments/engineering/dev/career-development/framework.md:456]: The typical leveling in Big Tech is Staff -> Sr. Staff -> Principal; maybe we can call it Sr. Staff+ (TBD) for now.

**Review by vrto — APPROVED** [review]
LGTM 👍 ✅

**Review by mrnugget — APPROVED** [review]

File: content/departments/engineering/dev/career-development/framework.md (hunk 1 above)
Comment (chrispine) [reply content/departments/engineering/dev/career-development/framework.md:456]: This was proposed by Quinn [here](https://sourcegraph.slack.com/archives/C03KFJTH5TN/p1658800158412539?thread_ts=1658504132.726919&cid=C03KFJTH5TN), and restated [here](https://docs.google.com/document/d/16fL4KETgtPPbhux9DkZC6hLbJg6qQO8bw2REuqA8qXs/edit#heading=h.gwxh3351wx5i), and no one commented on it, so I took that to mean we were all happy with this title.

If that is not the case, we can certainly discuss it more.

**Pull request merged**